.travis.yml

```diff
@@ -1,19 +1,22 @@
 matrix:
-    include:
-        - language: python
-          python: 3.6
-          sudo: false
-          before_install:
-              - cd python
-          install:
-              - pip install -r requirements.txt
-          script:
-              - python -m pytest
-        - language: cpp
-          sudo: false
-          script:
-              - mkdir build
-              - cd build
-              - cmake -DENABLE_CI_BUILD=ON ../
-              - cmake --build .
-              - ctest --verbose .
+  include:
+    - language: python
+      python: 3.6
+      sudo: false
+      before_install:
+        - cd python
+      install:
+        - pip install -r requirements.txt
+      script:
+        - python -m pytest
+    - language: python
+      python: 2.7
+      sudo: false
+      cache:
+        directories:
+          - "~/.platformio"
+      install:
+        - pip install -U platformio
+        - platformio update
+      script:
+        - platformio test --verbose --environment ci_native
```

Comment:
Can you add `platformio run` as part of the build script? This ensures that all code compiles, at least, before test cases are run

Reply:
There is no `main` or `setup` `loop` functions so the build will just fail. The test functions builds all code, however, and will spit out the build error if one occurs.